internal/controllers/core/podlogstream/podlogstreamcontroller.go

```diff
@@ -213,46 +231,50 @@ func (r *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			// The active pod watcher got canceled somehow,
 			// so we need to create a new one that picks up
 			// where it left off.
-			startWatchTime = <-existing.terminationTime
-			r.hasClosedStream[key] = true
-			if c.State.Terminated != nil {
-				r.mutateStatus(streamName, container.Name(c.Name), func(cs *ContainerLogStreamStatus) {
+			<-existing.doneCh
+			startWatchTime = existing.doneWatchTime
+			debounce = existing.debounce
+			c.hasClosedStream[key] = true
+			if co.State.Terminated != nil {
+				c.mutateContainerStatus(streamName, container.Name(co.Name), func(cs *ContainerLogStreamStatus) {
 					cs.Terminated = true
 					cs.Active = false
 					cs.Error = ""
 				})
 				continue
 			}
+
+			if c.clock.Since(existing.doneWatchTime) < debounce {
+				// TODO(nick): This should return a ctrl.Result that reschedules the reconcile.
```

Comment:
Are you planning this for future PR? (Fine either way with me, just making sure it's not an oversight)

Reply:
done! just needed to add some tests for it